packages/location-management/src/components/EditLocationUnit/index.tsx

```diff
@@ -151,37 +150,31 @@ const EditLocationUnit = (props: EditLocationUnitProps) => {
     }
   );
 
-  useQueries(
-    locationUnits.data
+  const treeDataQuery = useQueries(
+    locationUnits.data && locationUnits.data.length
       ? locationUnits.data.map((location) => {
           return {
             queryKey: [LOCATION_HIERARCHY, location.id],
             queryFn: () => new OpenSRPService(LOCATION_HIERARCHY, opensrpBaseURL).read(location.id),
             onError: () => sendErrorNotification(lang.ERROR_OCCURRED),
             // Todo : useQueries doesn't support select or types yet https://github.com/tannerlinsley/react-query/pull/1527
-            onSuccess: (res) => {
-              // the Tree is not already in the locationtree
-              if (!treeData.find((data) => JSON.stringify(data) === JSON.stringify(res))) {
-                // if the tree already exist i.e only a part of tree is changed, we need to replce old one else add new one
-                const alreadyExist = treeData.findIndex(
-                  (tree) => tree.id === (res as ParsedHierarchyNode).id
-                );
-
-                if (alreadyExist !== -1)
-                  treeData.splice(alreadyExist, 1, res as ParsedHierarchyNode);
-                else setTreeData((treeData) => [...treeData, res as ParsedHierarchyNode]);
-              }
-            },
-            // Todo : useQueries doesn't support select or types yet https://github.com/tannerlinsley/react-query/pull/1527
             select: (res) => generateJurisdictionTree(res as RawOpenSRPHierarchy).model,
           };
         })
       : []
-  );
-
-  if (loading || !locationUnits.data || locationUnits.data.length) {
+  ) as UseQueryResult<ParsedHierarchyNode>[];
+
+  const treeData = treeDataQuery
```

Comment:
I'm not sure about this but maybe we should store treeData in state here? I think this is where you use the use-deep-compare-effect for diffing